src/Microsoft.DotNet.Cli.Utils/CommandResolver.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using NuGet.Frameworks;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal static class CommandResolver
+    {
+        public static ProcessStartInfo TryResolvePathAndArgs(string commandName, string args, NuGetFramework framework = null)
```

Comment:
I added a CommandResolutionStrategy enum into Command. Since TryResolvePathAndArgs will compute it instead, I'm wondering what's the best way to pass it back to Command. I think you mentioned a struct with path, args, and resolution strategy at some point